contracts/script/multisigTransactionProposals/safeSDK/modifyStateHistoryRetentionPeriod.ts

```diff
@@ -0,0 +1,130 @@
+import dotenv from "dotenv";
+import { ethers } from "ethers";
+import { EthersAdapter } from "@safe-global/protocol-kit";
+import SafeApiKit from "@safe-global/api-kit";
+import Safe from "@safe-global/protocol-kit";
+import { getEnvVar, createSafeTransactionData, validateEthereumAddress } from "./utils";
+
+// declaring the type returned by the createTransaction method in the safe package locally (since the return type isn't exposed) so that if it's updated, it's reflected here too
+type LocalSafeTransaction = Awaited<ReturnType<Safe["createTransaction"]>>;
+
+async function main() {
+  dotenv.config();
+
+  try {
+    // Initialize web3 provider using the RPC URL from environment variables
+    const web3Provider = new ethers.JsonRpcProvider(getEnvVar("RPC_URL"));
+    // Create a signer using the orchestrator's private key and the web3 provider
+    const orchestratorSigner = new ethers.Wallet(getEnvVar("SAFE_ORCHESTRATOR_PRIVATE_KEY"), web3Provider);
+
+    // Set up Eth Adapter with ethers and the signer
+    const ethAdapter = new EthersAdapter({
+      ethers,
+      signerOrProvider: orchestratorSigner,
+    });
+
+    const chainId = await ethAdapter.getChainId();
+    const safeService = new SafeApiKit({ chainId });
+    const safeAddress = getEnvVar("SAFE_MULTISIG_ADDRESS");
+    validateEthereumAddress(safeAddress);
+    const safeSdk = await Safe.create({ ethAdapter, safeAddress });
+    const orchestratorSignerAddress = await orchestratorSigner.getAddress();
+
+    const stateHistoryRetentionPeriod = parseInt(getEnvVar("STATE_HISTORY_RETENTION_PERIOD"));
+    console.log("Modify the stateHistoryRetentionPeriod to ", stateHistoryRetentionPeriod);
+
+    await proposeSetStateHistoryRetentionTransaction(
+      safeSdk,
+      safeService,
+      orchestratorSignerAddress,
+      safeAddress,
+      stateHistoryRetentionPeriod,
+    );
+
+    console.log(
+      `The other owners of the Safe Multisig wallet need to sign the transaction via the Safe UI https://app.safe.global/transactions/queue?safe=sep:${safeAddress}`,
```

Comment:
nit: `Proposal created successfully. The other owners of the Safe Multisig wallet need to sign the transaction via the Safe UI https://app.safe.global/transactions/queue?safe=sep:${safeAddress}`

Reply:
thanks, addressed here https://github.com/EspressoSystems/espresso-sequencer/commit/a2ab0f54a13689832cadbd0500193e1cf3ebeb75#diff-ed9be146bb82143e9fdc2e1e482dde2e0977637db45081c1b52d42ad64ca578fR66